### ISSUE TITLE
Blurring cuts in and out

### ISSUE THREAD

**Issue by karulont** [karulont/i3lock-blur#5]
I received a "bug report".

> The blurring cuts in and out when I use this; one moment it's blurred, then the next it's clear, then it's blurred again, seeming to update in time with the clock on my statusbar. 
> - Celti

**Comment by bchretien** [comment]
I noticed the same "ticking" bug with the last release on Arch.

**Comment by karulont** [comment]
Are you running a composer?
What windowmanager are you using?
What Desktop environment?

**Comment by bchretien** [comment]
* Window manager: i3
* Compositor: compton
* No desktop environment
* Things running: i3blocks (for i3bar), tmux sessions inside termite, qutebrowser

I tried disabling i3blocks since this is the only thing "ticking" (the bar is
updated at some given intervals), and this bug stopped. I also observe some
visual artifacts when unlocking: some parts remain blurred (e.g. tab bar of my
browser, terminals etc.).

All of these bugs do not happen when the blurring feature is not
enabled.

**Comment by karulont** [comment]
i3blocks probably just updates the screen, which causes a redraw. You can put something other to the background, for example leave a video playing. That should rule out i3blocks.

I think that the problem is with compton. i3lock-blur uses the same overlay window as compton.

I got some university stuff going on right now, but I will find time next week, to look into it. And test it with compton myself.

**Comment by bchretien** [comment]
Indeed, confirmed with a video, and killing compton solves it. You can find my compton configuration file  [here](https://github.com/bchretien/i3-config/blob/master/compton.conf).

**Comment by karulont** [comment]
I did a little work on it, will continue tomorrow. I try to use screensaver window instead of composite overlay window.

Currently the problem is, that compton is a composite manager and uses composite overlay window, and i3lock-blur tries to use the same window, which causes a conflict between them.

**Comment by Tasqa** [comment]
Is there any news on the progress of this bug? I'm running into the same issue now?

**Comment by karulont** [comment]
I have to admit, that I have been a little lazy.
I started working on it, then got into some schoolwork and forgot about it.
Unfortunately I am too busy right now to start working on it.

I guess that you also use a composite manager. Currently i3lock-blur will not work correctly when using a composite manager because they try to draw onto the same window and therefore are in conflict.

**Comment by Tasqa** [comment]
Allright, thanks for the update. Then I'll just use different locking options for now.

**Comment by karulont** [comment]
Use a wrapper script that first kills your composer, runs i3lock-blur, then starts your composer. This is probably the easiest solution.

**Comment by Reventl0v** [comment]
I think adding "paint-on-overlay = false;" to your compton config (~/.config/compton.conf) makes it.

R.

**Comment by megawebmaster** [comment]
I've got another proposal - why do we bother to update pixmaps if we can get a blurred background once and display it? This should fix #10 as well. Is it worth working on? :)

**Comment by karulont** [comment]
The whole idea of i3lock-blur is to update the blurred image when the background changes. If you want to blur only once, then you can use [this](https://gist.github.com/csivanich/10914698) -- it was the inspiration for writing i3lock-blur.

For me it felt awkward that I see something on the lock screen, open the lock, and something completely different is actually on the screen.

**Comment by megawebmaster** [comment]
Well, the problem with the script is that it works so slow it's unusable. On modern, powerful PC it takes 2 seconds to lock and that's unacceptable. 

Unfortunately for now (even with compton turned off) the screen is weirdly flickering with i3lock-blur so I will probably create a fork and figure out how to make it single time blurring. IMO when locked screen does not change that often or you cannot really see what changed after opening it up. Maybe another switch for i3lock would be good?

**Comment by karulont** [comment]
You raised a valid argument against the script. Therefore I thought I would give it a try. Check out blur-once branch and let me know if it works for you.

**Comment by karulont** [comment]
I forgot to add that, when trying out blur-once branch, you should use the options: i3lock -fo.

**Comment by megawebmaster** [comment]
Thanks for taking your time to implement my suggestion! It works perfectly, even with Compton running, now I can easily use it as my lock screen! Thanks again!

**Comment by TomokoK** [comment]
Has there been any progress on this bug at all? It looks like you removed the --once flag, as it doesn't work on my end. Without it I have horrible flickering with Compton.
